metis-indexing/src/test/java/eu/europeana/indexing/fullbean/RdfToFullBeanConverterTest.java

```diff
@@ -0,0 +1,142 @@
+package eu.europeana.indexing.fullbean;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import eu.europeana.corelib.solr.bean.impl.FullBeanImpl;
+import eu.europeana.indexing.base.IndexingTestUtils;
+import eu.europeana.indexing.utils.RdfWrapper;
+import eu.europeana.metis.schema.convert.RdfConversionUtils;
+import eu.europeana.metis.schema.convert.SerializationException;
+import org.junit.jupiter.api.Test;
+
+class RdfToFullBeanConverterTest {
+
+  @Test
+  void convertRdfToFullBean() throws SerializationException {
+    RdfToFullBeanConverter rdfToFullBeanConverter = new RdfToFullBeanConverter();
+    final RdfConversionUtils conversionUtils = new RdfConversionUtils();
+    final RdfWrapper inputRdf = new RdfWrapper(conversionUtils.convertStringToRdf(
+        IndexingTestUtils.getResourceFileContent("europeana_record_rdf.xml")));
+
+    FullBeanImpl fullBean = rdfToFullBeanConverter.convertRdfToFullBean(inputRdf);
+
+    assertAbout(fullBean);
+    assertAgents(fullBean);
+    assertAggregations(fullBean);
+    assertConcepts(fullBean);
+    assertEuropeanaAggregation(fullBean);
+    assertEuropeanaCollection(fullBean);
+    assertLicenses(fullBean);
+    assertOrganizations(fullBean);
+    assertPlaces(fullBean);
+    assertProvidedCHOs(fullBean);
+    assertProxies(fullBean);
+    assertQualityAnnotations(fullBean);
+    assertServices(fullBean);
+    assertSizes(fullBean);
+    assertTimeSpans(fullBean);
+    assertTimestamps(fullBean);
+  }
+
+  private static void assertTimestamps(FullBeanImpl fullBean) {
+    assertEquals("1970-01-01T00:00:00Z", fullBean.getTimestampCreated().toInstant().toString());
+    assertEquals("1970-01-01T00:00:00Z", fullBean.getTimestampUpdated().toInstant().toString());
+  }
+
+  private static void assertAbout(FullBeanImpl fullBean) {
+    assertEquals("/277/CMC_HA_1185", fullBean.getAbout());
+  }
+
+  private static void assertEuropeanaCollection(FullBeanImpl fullBean) {
+    assertEquals("277_local_09012024_1543", fullBean.getEuropeanaCollectionName()[0]);
+  }
+
+  private static void assertProxies(FullBeanImpl fullBean) {
+    assertEquals("/proxy/europeana/277/CMC_HA_1185", fullBean.getProxies().get(0).getAbout());
```

Comment:
get[x] and/or getFirst does not seem like a sound check. The order might be different on a second execution. Probably it's better to get if it's part of the list.
This involves all the cases in this class

Reply:
Refactored.